packages/ra-ui-materialui/src/input/CheckboxGroupInput.tsx

```diff
@@ -184,7 +186,7 @@ export const CheckboxGroupInput: FunctionComponent<CheckboxGroupInputProps> = pr
         [allChoices, formOnChange, formOnBlur, optionValue, value]
     );
 
-    if (isLoading && (!allChoices || allChoices.length === 0)) {
+    if (isPending && (!allChoices || allChoices.length === 0)) {
```

Comment:
Shouldn't we use `isPending || isLoading` instead of `isPending` for backwards compatibility?

(same question applies to other similar inputs)

Reply:
`isLoading` implies `isPending`

Reply:
And no, we want to follow the react query practices anyway

Reply:
Right, I got confused by spending too much time on this, sorry about that :sweat_smile: 
Resuming the review...